controllers/btpoperator_controller.go

```diff
@@ -520,22 +536,150 @@ func (r *BtpOperatorReconciler) reconcileResources(ctx context.Context, s *corev
 	return nil
 }
 
-func (r *BtpOperatorReconciler) getResourcesToApplyPath() string {
-	return fmt.Sprintf("%s%capply", ResourcesPath, os.PathSeparator)
+func (r *BtpOperatorReconciler) handleSecretChanges(ctx context.Context, resourcesToApply []*unstructured.Unstructured, managerSecret *corev1.Secret, logger *logr.Logger) error {
+	// this is first scenario when provisoning. We cannot restart operator deployment here since resources are not yet applied
+	operator := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      DeploymentName,
+			Namespace: ChartNamespace,
+		},
+	}
+	if err := r.Get(ctx, client.ObjectKeyFromObject(operator), operator); err != nil {
+		if k8serrors.IsNotFound(err) {
+			logger.Info("operator deployment not found")
+			return nil
+		}
+		logger.Error(err, "while getting operator deployment")
+		return fmt.Errorf("while operator deployment: %w", err)
+	}
+
+	configMapIndex, secretIndex := r.GetConfigMapAndSecretIndexes(resourcesToApply)
+	configMap := resourcesToApply[configMapIndex]
+	operatorSecret := resourcesToApply[secretIndex]
+
+	operatorNeedRestart, err := r.verifySecretChanges(ctx, configMap, operatorSecret, managerSecret, logger)
+	if err != nil {
+		return fmt.Errorf("failed to reconcile manager secret changes: %w", err)
+	}
+	if operatorNeedRestart {
+		logger.Info("need to restart operator")
+		logger.Info("deleting secret " + clusterIdSecretName + " to restart operator deployment")
+		err := r.deleteClusterIDSecret(ctx, logger)
+		if err != nil {
+			return fmt.Errorf("failed to delete secret %s : %w", clusterIdSecretName, err)
+		}
+		logger.Info("secret " + clusterIdSecretName + " deleted")
+		logger.Info("restarting operator deployment")
+		if err := r.restartOperatorDeployment(ctx); err != nil {
+			return fmt.Errorf("failed to restart operator deployment: %w", err)
+		}
+		logger.Info("operator deployment restarted")
+	}
+	return nil
 }
 
-func (r *BtpOperatorReconciler) prepareModuleResourcesFromManifests(ctx context.Context, resourcesToApply []*unstructured.Unstructured, s *corev1.Secret) error {
-	logger := log.FromContext(ctx)
+func (r *BtpOperatorReconciler) verifySecretChanges(ctx context.Context, futureConfigMap, operatorSecret *unstructured.Unstructured, managerSecret *corev1.Secret, logger *logr.Logger) (bool, error) {
```

Comment:
It more checking if we are not in sync (secret vs config map and secret).
Or if we need reconciliation of `sap-btp-service-operator`, `sap-btp-operator-config` based on values in `sap-btp-manager`.